queue/driver_mongo.go

```diff
@@ -495,32 +495,72 @@ func (d *mongoDriver) Get(ctx context.Context, name string) (amboy.Job, error) {
 	return output, nil
 }
 
-func (d *mongoDriver) GetAttempt(ctx context.Context, name string, attempt int) (amboy.Job, error) {
+// GetAttempt finds a retryable job matching the given job ID and execution
+// attempt. This returns an error if no matching job is found.
+func (d *mongoDriver) GetAttempt(ctx context.Context, id string, attempt int) (amboy.Job, error) {
 	matchIDAndAttempt := bson.M{
-		"retry_info.base_job_id":     name,
+		"retry_info.base_job_id":     id,
```

Comment:
We should add an index  `retry_info.base_job_id: 1, retry_info.current_attempt: 1` for this and `GetAllAttempts`.

Reply:
Hmm, it might be easier to just use the existing index with `{retry_info.base_job_id: 1, retry_info.current_attempt: -1}`. I modified the logic to make it ascending order in-memory.

Reply:
HA I missed that one! I didn't look up far enough in the file. Sorry.